src/common/components/Input.test.tsx

```diff
@@ -0,0 +1,105 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { ErrorBoundary } from 'react-error-boundary';
+import { Provider } from 'react-redux';
+import { createTestStore } from '../../app/store';
+import ErrorPage from '../../features/layout/ErrorPage';
+import { Input } from './Input';
+import { handlerPropError, missingLabelFactory } from './Input.common';
+import classes from './Input.module.scss';
+
+const consoleError = jest.spyOn(console, 'error');
+// This mockImplementation supresses console.error calls.
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+consoleError.mockImplementation(() => {});
+
+describe('Input related components', () => {
+  afterAll(() => {
+    consoleError.mockRestore();
+  });
+
+  afterEach(() => {
+    consoleError.mockClear();
+  });
+
+  test('Input Renders', () => {
+    render(<Input idForLabel={'test'} label={<>test</>} />);
+    const inputText = screen.getByLabelText('test', { exact: false });
+    expect(inputText).toBeInTheDocument();
+  });
+
+  test('Specifying changeHandler and validate throws an error', () => {
```

Comment:
This is a neat testing strategy, but I think more precisely typing <Input/> will eliminate the need for it here. (Sorry 🙈 I'm only just now understanding what you meant about errors inside the component)

Reply:
Using `react-hook-form` made the tests simpler and dispensed with the need for this kind of error handling at this time. We might use this pattern at some other point in a separate context where errors must be tested.